EIPS/eip-7886.md

```diff
@@ -0,0 +1,400 @@
+
+## Specification
+
+### Change to the header structure
+
+#### Deferred fields 
+
+In order to be verifiable before execution, the header cannot commit to any execution output. In particular, we need to defer these fields: `state_root, receipt_root, bloom, gas_used, requests_hash`. We replace them with the equivalent execution outputs from the parent block. 
+
+#### Coinbase signature over the header
+
+We include a signature from `COINBASE` over the rest of the header in the header , so that the `COINBASE` address can authorize the upfront payment of inclusion costs.
+
+The final header structure then is: 
+
+```python
+class Header:
+    parent_hash: Hash32
+    ommers_hash: Hash32
+    coinbase: Address
+    pre_state_root: Root # Deferred
+    transactions_root: Root
+    parent_receipt_root: Root # Deferred
+    parent_bloom: Bloom # Deferred
+    difficulty: Uint
+    number: Uint
+    gas_limit: Uint
+    parent_gas_used: Uint
+    timestamp: U256
+    extra_data: Bytes
+    prev_randao: Bytes32
+    nonce: Bytes8
+    base_fee_per_gas: Uint
+    withdrawals_root: Root
+    blob_gas_used: U64
+    excess_blob_gas: U64
+    parent_beacon_block_root: Root
+    parent_requests_hash: Hash32 # Deferred
+    # Signature over the header
+    y_parity: U256
+    r: U256
+    s: U256
+```
+
+
+The following function is used to recover the signer’s address, intended to be `COINBASE`, from the signed block header:
+
+```python
+def recover_header_signer(
+    chain_id: U64,
+    header: Header,
+) -> Address:
+    signing_hash = keccak256(
+        b"0x06"
+        + rlp.encode(
+            (
+                chain_id,
+                header.parent_hash,
+                header.ommers_hash,
+                header.coinbase,
+                header.pre_state_root,
+                header.transactions_root,
+                header.parent_receipt_root,
+                header.parent_bloom,
+                header.difficulty,
+                header.number,
+                header.gas_limit,
+                header.parent_gas_used,
+                header.timestamp,
+                header.extra_data,
+                header.prev_randao,
+                header.nonce,
+                header.base_fee_per_gas,
+                header.withdrawals_root,
+                header.blob_gas_used,
+                header.excess_blob_gas,
+                header.parent_beacon_block_root,
+                header.parent_requests_hash,
+            )
+        )
+    )
+```
+
+By signing over the header, the `COINBASE` address explicitly accepts responsibility for the upfront inclusion costs *of this block*. Therefore, the recovered address MUST equal the block's `COINBASE`. The `COINBASE`'s commitment is protected from replay attacks, because the header is a commitment to the block, so the signature only serves as an authorization for the exact block for which the `COINBASE` has agreed to take responsibility.
```

Comment:
Little bit more rationale in here. I'm going to stop commenting on this; you get the idea.